### ISSUE TITLE
`currentParsedContent` returns a first-char-repeated string from `ByteArrayInputStream` with `setReadInputOnSeparateThread` disabled

### ISSUE THREAD

**Issue by HyukjinKwon** [uniVocity/univocity-parsers#185]
Reproducer:

```java
import java.io.ByteArrayInputStream;

import com.univocity.parsers.csv.CsvParser;
import com.univocity.parsers.csv.CsvParserSettings;

class Main {
    public static void main(String[] args) throws Exception {
        CsvParserSettings settings = new CsvParserSettings();
        settings.setReadInputOnSeparateThread(false);
        CsvParser parser = new CsvParser(settings);

        parser.beginParsing(new ByteArrayInputStream("a".getBytes("UTF-8")));
        parser.parseNext();
        System.out.println(parser.getContext().currentParsedContent());
    }
}
```

This produces `aa` whereas:

```java
import java.io.StringReader;

import com.univocity.parsers.csv.CsvParser;
import com.univocity.parsers.csv.CsvParserSettings;

class Main {
    public static void main(String[] args) throws Exception {
        CsvParserSettings settings = new CsvParserSettings();
        settings.setReadInputOnSeparateThread(false);
        CsvParser parser = new CsvParser(settings);

        parser.beginParsing(new StringReader("a"));
        parser.parseNext();
        System.out.println(parser.getContext().currentParsedContent());
    }
}
```

produces `a`. The diff is, `ByteArrayInputStream` vs `StringReader`.

I expected the same results but this was different. The first case looks a minor regression somewhere between 2.5.0 and 2.5.1 given my manual tests.

The first case produced `a` with 2.5.0.

**Comment by jbax** [comment]
Thank you for reporting the bug! I've fixed it and released a 2.5.4-SNAPSHOT build. I'm planning to release the final 2.5.4 build this weekend so if you could confirm it's working for you that would be great.

Cheers!

**Comment by HyukjinKwon** [comment]
Thank you for the quick action. I checked this issue is fixed.
